docs/guides/testing.md

```diff
@@ -16,45 +16,57 @@ Nexus comes with a special testing module that you can import from `nexus/testin
 Before jumping into test suites we will wrap the `createTestContext` with a pattern that more tightly integrates it into `jest`. Nexus will probably ship something like as follows or better in the future, but for now you can copy this into your projects:
 
 ```ts
-// tests/__helpers.ts
-import { createTestContext as originalCreateTestContext, TestContext } from 'nexus/testing'
-
-export function createTestContext(): TestContext {
-  let ctx: TestContext
-
-  beforeAll(async () => {
-    ctx = await createTestContext()
-    await ctx.app.server.start()
-  })
-
-  afterAll(async () => {
-    await ctx.app.server.stop()
-  })
-
-  // @ts-ignore
-  return ctx
+// tests/environment.js
+const NodeEnvironment = require('jest-environment-node')
+const { createTestContext } = require('nexus/testing')
+
+module.exports = class CustomEnvironment extends NodeEnvironment {
+  nexus = null
+
+  async setup() {
+    await super.setup()
+    this.nexus = await createTestContext()
+    await this.nexus.app.server.start()
+    this.global.nexus = this.nexus
+  }
+
+  async teardown() {
+    await super.teardown()
+    await this.nexus.app.server.stop()
+  }
```

Comment:
What are the reasons we want to move to a jest environment for that? Makes use of javascript + global type declaration + jest config, I feel like it adds a lot more indirection for simple use-cases 🤔

Reply:
What if we just ship a Jest environment?

```js
{
  jest: {
    environment: "nexus/testsing/jest/environment"
  }
}
```